src/GeneralStateTestsFiller/stCreateTest/CreateResultsFiller.yml

```diff
@@ -0,0 +1,327 @@
+      code:   | 
+        {
+
+          ; Variables are 0x20 bytes (= 256 bits) apart, except for
+          ; code buffers that get 0x100 (256 bytes)
+          (def 'creation          0x100)
+          (def 'callType          0x120)
+          (def 'constructor       0x140)
+          (def 'contractCode      0x200)
+          (def 'constructorCode   0x300)
+          (def 'extCode           0x400)
+          (def 'contractLength    0x520)
+          (def 'constructorLength 0x540)
+          (def 'extLength         0x560)
+          (def 'addr1             0x600)
+          (def 'addr2             0x620)
+          (def 'callRet           0x640)
+
+          ; Other constants
+          (def 'NOP 0)   ; No OPeration
+
+          ; Understand the input.
+          [creation]       (shr $ 0 248)
+          [callType]       (and (shr $ 0 240) 0xFF)
+          [constructor]    (and (shr $ 0 232) 0xFF)
+
+          ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+          ; Create the contract and a constructor to pass to CREATE[2]
+          ;
+
+
+          (if (= @constructor 0) 
+            [constructorLength] 
+              (lll {   ; successful constructor
+                 (return contractCode
+                    ; write the contract to constructor memory 
+                    (lll
+                       (call 0xFFFF 0x60A7 0 0 0 0 0)
+                       contractCode
+                    ) ; inner lll
+                 )  } ; return contractCode (lll ...)
+                  constructorCode
+              )       ; outer lll, if successful
+              NOP
+            )
+
+          (if (= @constructor 1) 
+            [constructorLength] 
+              (lll {   ; constructor if out of gas
+                 (KECCAK256 0x00 0x2fffff)
+                 (return contractCode
+                    ; write the contract to constructor memory 
+                    (lll
+                       (call 0xFFFF 0x60A7 0 0 0 0 0)
+                       contractCode
+                    ) ; inner lll
+                 )  } ; return contractCode (lll ...)
+                  constructorCode
+              )       ; outer lll, if out of gas
+              NOP
+            )
+
+
+          (if (= @constructor 2) 
+            [constructorLength] 
+              (lll {   ; constructor if revert
+                 (revert 0 0)
+                 (return contractCode
+                    ; write the contract to constructor memory 
+                    (lll
+                       (call 0xFFFF 0x60A7 0 0 0 0 0)
+                       contractCode
+                    ) ; inner lll
+                 )  } ; return contractCode (lll ...)
+                  constructorCode
+              )       ; outer lll, revert
+              NOP
+            )
+
+
+          ; Write the contract to our own memory (as the 0xcc...cc contract)
+          [contractLength] (lll 
+             (call 0xFFFF 0x60A7 0 0 0 0 0)
```

Comment:
what is this call  ?

Reply:
Same as above:

No. What happens is that we need the contract code in two places with different memories:

The constructor, to create the new contract
The LLL contract, to verify that EXTCODECOPY returns the same values as the ones we sent
The inner lll inside the constructor creates the code in the constructor memory. This lll creates the code in our contract's memory.

Reply:
But why is the call in constructor?

Reply:
It isn't in the constructor. It's in the code of the contract being constructed. Anyway, that will be a lot clearer in the next version, where it's all hidden in a macro.